external-crates/move/crates/move-analyzer/trace-debug/src/extension.ts

```diff
@@ -427,21 +443,34 @@ function getTracedFunctionInfo(tracePath: string): TracedFunctionInfo {
  * Given trace file path and traced function, constructs a string of the form
  * `<package>::<module>::<function_name>`, taking package from the trace file name
  * (module name and function are the same in the file name and in the trace itself).
+ * We recognize two types of traces and corresponding file names as well as directory
+ * structure: generated when running unit tests and generated by the replay tool.
  *
  * @param tracePath path to the trace file.
  * @param tracedFunctionInfo traced function info.
  * @returns string of the form `<package>::<module>::<function_name>`.
  */
 function constructTraceInfo(tracePath: string, tracedFunctionInfo: TracedFunctionInfo): string {
     const tracedFileBaseName = path.basename(tracePath, path.extname(tracePath));
-    const fileBaseNameSuffix = '__' + tracedFunctionInfo.module + '__' + tracedFunctionInfo.function;
-    if (!tracedFileBaseName.endsWith(fileBaseNameSuffix)) {
-        throw new Error('Trace file name (' + tracedFileBaseName + ')'
-            + 'does not end with expected suffix (' + fileBaseNameSuffix + ')'
-            + ' obtained from concateneting module and entry function found in the trace');
+    if (tracedFileBaseName === REPLAY_TRACE_FILE_NAME) {
```

Comment:
Yea, I think there's a discussion to be had here about how to handle naming of tracing/replay-related metadata for a specific transaction replay since e.g, I could see it getting very confusing if you've replayed multiple transactions and you aren't sure which trace you're dealing with (or gas profiling info etc).

Reply:
New naming proposal in in the other [comment](https://github.com/MystenLabs/sui/pull/21241#discussion_r1960312109)